lib/promotion.rb

```diff
@@ -121,21 +121,25 @@ def self.tl3_lost?(user)
   end
 
   # Figure out what a user's trust level should be from scratch
-  def self.recalculate(user, performed_by = nil)
+  def self.recalculate(user, performed_by = nil, set_previous_trust_level: false)
```

Comment:
This is a little of a side track, but when I reviewed the code I'd thought: "Why am I recalculating my promotion instead of my trust level?" because of the invocation `Promotion.recalculate()`.

How do you feel about adding a method in `trust_level.rb` which returns the calculated trust level without side effects (essentially the existing calculation part)?

```
  def self.calculate(user)
    # First, use the manual locked level
    unless user.manual_locked_trust_level.nil?
      return user.manual_locked_trust_level
    end

    # Then consider the group locked level
    # or the previous trust level
    granted_trust_level = user.group_granted_trust_level
    previous_trust_level = find_previous_trust_level(user)

    if !previous_trust_level && !granted_trust_level
      return 0
    end

    if !previous_trust_level
      return granted_trust_level
    end

    if !granted_trust_level
      return previous_trust_level
    end

    (previous_trust_level > granted_trust_level) ? previous_trust_level : granted_trust_level
  end
```